crates/storage/store_db/libmdbx.rs

```diff
@@ -169,13 +169,23 @@ impl StoreEngine for Store {
 
                 tx.upsert::<Bodies>(
                     hash.into(),
-                    BlockBodyRLP::from_bytes(block.body.encode_to_vec()),
+                    BlockBodyRLP::from_bytes(
```

Comment:
here you are leaking an abstraction. Why can't you have `encode_to_vec` use the cached version?